apache/superset
# feat(sqllab): save query parameters in database

### SUMMARY

* Saves template parameters when query is saved.
* Updates frontend to use v1 saved query rest api, 
* Removes unnecessary parameters from the payload which api doesn't need.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF


### TESTING INSTRUCTIONS

1. Enable `ENABLE_TEMPLATE_PROCESSING` flag
2. write any query in sqllab with params eg.: `SELECT '{{ p1 }}'` and set parameters `{"p1": "value from params"}`
3. save the query and **close the tab** then go to saved queries
4. when saved query is opened, paramter values should show up exactly as it was saved, and when query is run it should show 1 column with value `value from params`
5. updating the query with sql or parameters should correctly update the query.
6. also test changing the query name, and it should update tab name.

### ADDITIONAL INFORMATION


- [x] Has associated issue: #20576
  - [ ] Runtime estimates and downtime expectations provided -> **not needed, cuz it just adds 1 column without default value**
